app/controllers/v1/flavours_controller.rb

```diff
@@ -5,17 +5,18 @@ class FlavoursController < ApplicationController
     def index
       flavours = Flavours.new
       flavour_list = flavours.list(params[:appliance_id],params[:site_id])
-      flavour_list.blank? ? respond_with(flavour_list, status: 204) : respond_with(flavour_list)#[0, @limit])
+      if flavour_list.first.key? (:message)
```

Comment:
You are checking the blankness of `flavour_list` on L11 ... but on L8 you are reaching directly into its content. What will `.key?` do when `flavour_list.first` returns `nil`? I suspect it will raise an error (NoMethodError on NilClass). Can you check that? Maybe change the branch order here?